build/colmena.nix

```diff
@@ -0,0 +1,74 @@
+# heavily adapted from https://github.com/juspay/colmena-flake
+# Original license: GNU Affero General Public License v3.0
+{
+  config,
+  lib,
+  self,
+  inputs,
+  ...
+}:
+{
+  options.colmena = {
+    hosts = lib.mkOption {
+      type = lib.types.attrsOf (
+        lib.types.submodule (
+          { name, ... }:
+          {
+            options = {
+              targetHost = lib.mkOption {
+                type = lib.types.str;
+                default = "${name}.nixos.org";
```

Comment:
Do our machines have a fqdn configured that we could/should use instead?

Reply:
Feels a bit like a layer violation to access the nixos module system from the flake parts module system. It might make parallel evaluation harder.

Reply:
Sure. Perhaps this is a colmena question instead.